### DISCUSSION TITLE
I have created a network thread X.

### DISCUSSION THREAD

**Discussion by xuyun018** [fltk/fltk#1040] [Q&A] [answered]
I have created a network thread X, and when it receives data, I want the FLTK interface main thread Y to process it. How do I do it?

**Comment by MatthiasWM** [comment, marked as answer]
You can use `Fl::lock()` and `Fl::unlock()` to access FLTK from a thread for everything but openeing and closing windows. You can also use `Fl::awake()` with an wake handler. The docs for all that are here: https://www.fltk.org/doc-1.4/advanced.html

**Comment by MatthiasWM** [comment]
Also note that you can monitor network sockets with `Fl::add_fd()` (https://www.fltk.org/doc-1.4/classFl.html#a8905e303e45e846b8ce77168f4cf7afe)

**Reply by xuyun018** [reply]
Tks

**Comment by Albrecht-S** [comment]
Note: moved from "Development" to "Q&A".

Please see chapter "[Advanced FLTK](https://www.fltk.org/doc-1.4/advanced.html)" in our docs for more info on FLTK + Threads (as Matthias wrote already, just a minute or two ago).

We recommend the approach with an [awake handler](https://www.fltk.org/doc-1.4/group__fl__multithread.html#ga22a404bcaf6641369e0725627d881556) because this is the most reliable way to process your thread data.

**Reply by xuyun018** [reply]
Tks!

**Comment by Albrecht-S** [comment]
@xuyun018 Has your question been answered? If yes, please mark one of the replies as answer and close this Discussion. Thanks.
